lib/util/documentvisibilitymonitor.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+/**
+ * The {@link DocumentVisibilityMonitor} monitors the visibility state of the DOM
+ * and executes the attached listeners in phase order when the DOM is visible.
+ */
+class DocumentVisibilityMonitor {
+  /**
+   * Constructor.
+   * @param {number} [nPhases=1] - the number of phases
+   */
+  constructor(nPhases = 1) {
+    Object.defineProperties(this, {
+      _listeners: {
+        value: []
+      },
+      _onVisibilityChange: {
+        value: () => {
+          if (document.visibilityState === 'visible') {
+            this._emitVisible();
+          }
+        }
+      }
+    });
+
+    for (let i = 0; i < nPhases; i++) {
+      this._listeners.push([]);
+    }
+  }
+
+  _listenerCount() {
```

Comment:
I think this can be made into a getter property
```suggestion
  get _listenerCount() {
```

Reply:
maybe we should talk more about this - but I think functions are in general better choice than properties,
 because 
1) they make it obvious that some processing in involved 
2) it avoids bugs like
``` 
if (foo._llistenerCount) {
  // do something useful
}
```
where the property does not really exist...but code wont fail immediately, thus causing bugs. 

3) they are bit easier to work with when testing/spying/stubbing on them